cpu/samd5x/cpu.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_samd5x
+ * @{
+ *
+ * @file        cpu.c
+ * @brief       Implementation of the CPU initialization for Atmel SAMD5x/SAME5x MCUs
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ * @}
+ */
+
+#include "cpu.h"
+#include "periph_conf.h"
+#include "periph/init.h"
+
+static void xosc32k_init(void)
+{
+    OSC32KCTRL->XOSC32K.reg = OSC32KCTRL_XOSC32K_ENABLE | OSC32KCTRL_XOSC32K_XTALEN
+                | OSC32KCTRL_XOSC32K_EN32K | OSC32KCTRL_XOSC32K_RUNSTDBY
+                | OSC32KCTRL_XOSC32K_STARTUP(7);
+
+    while (!OSC32KCTRL->STATUS.bit.XOSC32KRDY) {}
+
+    GCLK->GENCTRL[1].reg = GCLK_GENCTRL_SRC(GCLK_SOURCE_XOSC32K)
+                 | GCLK_GENCTRL_RUNSTDBY | GCLK_GENCTRL_GENEN;
+
+    GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL0].reg = GCLK_PCHCTRL_GEN(1)
+                          | GCLK_PCHCTRL_CHEN;
+    while (!(GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL0].reg & GCLK_PCHCTRL_CHEN)) {}
+
+    GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL032K].reg = GCLK_PCHCTRL_GEN(1)
+                             | GCLK_PCHCTRL_CHEN;
+    while (!(GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL032K].reg & GCLK_PCHCTRL_CHEN)) {}
+}
+
+static void dpll0_init(uint32_t f_cpu)
+{
+    const uint32_t LDR = ((f_cpu * 16) / 32768);
+
+    OSCCTRL->Dpll[0].DPLLRATIO.reg = OSCCTRL_DPLLRATIO_LDRFRAC(LDR % 16)
+                       | OSCCTRL_DPLLRATIO_LDR((LDR / 16) - 1);
+
+    OSCCTRL->Dpll[0].DPLLCTRLB.reg = OSCCTRL_DPLLCTRLB_REFCLK_XOSC32
+                       | OSCCTRL_DPLLCTRLB_DIV(1)
```

Comment:
Not needed.

Reply:
It doesn't hurt either - and I'd rather keep it as it makes the usage of this clearer.

Reply:
Ah! Ok, then you are right :)